clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -116,9 +116,9 @@
     public void seekToEnd(Collection<TopicPartition> partitions);
 
     /**
-     * @see KafkaConsumer#position(TopicPartition)
+     * @see KafkaConsumer#position(TopicPartition, long)
      */
-    public long position(TopicPartition partition);
+    public long position(TopicPartition partition, long timeout);
```

Comment:
Actually when I was discussing with @huxihx on the JIRA itself what I was thinking is the following:

1. The call traces of `Fetcher#resetOffsets` which use `INT_MAX` are from two public APIs: 

```
Fetcher#resetOffsets -> 
Fetcher#updateFetchPositions / Fetcher#resetOffsetsIfNeeded ->
KafkaConsumer#updateFetchPositions ->
KafkaConsumer#poll(timeout) / KafkaConsumer#position()
```

So we should pass in a `remaining` in the internal call from `KafkaConsumer#updateFetchPositions` all the way down to `Fetcher#resetOffsets` to replace `INT_MAX`; 

As for the public APIs, `poll` already have a timeout, we just need to let its internal `pollOnce` to pass in the updated `remaining` to `updateFetchPositions`, for `position` since it is defined as a non-blocking call, we should pass `0` to `updateFetchPositions` and if the call fails to return the offset in a single trial, it means no offset is cached locally and the remote offset request failes as well, we can throw `InvalidOffsetException` directly.

By doing this we do not need a KIP. Admittedly it may not be optimal to treat the non-blocking `position()` call like this, but this is to adherent with its semantics; if we do want to change its public behavior then I'd prefer option 1) from @hachikuji to modify the semantics of "non-blocking" calls of consumer as we did for producer.